applications/conceptual-model-editor/src/app/diagram/reactflow/components/menu-options.tsx

```diff
@@ -43,40 +45,74 @@ export const useMenuOptions = () => {
         deleteHandler?: () => void;
         position?: string;
     }) => {
-        const { openDetailHandler, createProfileHandler, modifyHandler, removeFromViewHandler, deleteHandler } = props;
+        const { openDetailHandler, createProfileHandler, modifyHandler, removeFromViewHandler, deleteHandler, position } = props;
+        // You have to actually define the type here explictly, typescript can't interfere the type correctly
+        const handlersWithTexts: Array<[(() => void) | undefined, string]> = [      
+            [openDetailHandler, "ℹ Detail"],
+            [modifyHandler, "✏ Modify"],
+            [undefined, "HorizontalSeparator"],
+            [createProfileHandler, "🧲 Create profile"],            
+            [undefined, "HorizontalSeparator"],
+            [removeFromViewHandler, "🕶 Remove from view"],
+            [deleteHandler, "🗑 Delete"],
+        ];
+        return MenuOptionsGeneral({handlersWithTexts, positionTailwind: position});        
+    };    
+
+    /**
+     * 
+     * @param props is an object with following properties:
+     * - handlersWithTexts is an array of tuples. The first property specifies the handler called on click and the second one the name. 
+     * Special case is if the handler is undefined and the name is "HorizontalSeparator".
+     * - positionTailwind specifies the position in tailwind notation.
+     * - positionCSS specifies the position in CSS notation. If both the tailwind and CSS positions are specified, the behavior is undefined.
+     * @returns
+     */
+    const MenuOptionsGeneral = (props: {
```

Comment:
Also since you need to comment on each property .. I would extract this into a type and add comments there:
```
type MenuOptionsGeneralType = {
 /** The first property specifies the handler called on click and the second one the name. ... */
 handlersWithTexts: Array<[(() => void) | undefined, string]>;
}
```
Also there is no need to repeat the types in the comment, for handlersWithTexts, as we are using TypeScript.